onnxruntime/core/providers/cpu/ml/tree_ensemble_aggregator.h

```diff
@@ -0,0 +1,481 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include "core/common/common.h"
+#include "core/framework/op_kernel.h"
+#include "ml_common.h"
+#include <math.h>
+
+namespace onnxruntime {
+namespace ml {
+
+struct TreeNodeElementId {
+  int tree_id;
+  int node_id;
+  inline bool operator==(const TreeNodeElementId& xyz) const {
+    return (tree_id == xyz.tree_id) && (node_id == xyz.node_id);
+  }
+  inline bool operator<(const TreeNodeElementId& xyz) const {
+    return ((tree_id < xyz.tree_id) || (tree_id == xyz.tree_id && node_id < xyz.node_id));
+  }
+};
+
+template <typename T>
+struct SparseValue {
+  int64_t i;
+  T value;
+};
+
+enum MissingTrack {
+  NONE,
+  TRUE,
+  FALSE
+};
+
+template <typename T>
+struct TreeNodeElement {
+  TreeNodeElementId id;
+  int feature_id;
+  T value;
+  T hitrates;
+  NODE_MODE mode;
+  TreeNodeElement<T>* truenode;
+  TreeNodeElement<T>* falsenode;
+  MissingTrack missing_tracks;
+  std::vector<SparseValue<T>> weights;
+
+  bool is_not_leave;
+  bool is_missing_track_true;
+};
+
+template <typename ITYPE, typename OTYPE>
+class _Aggregator {
+ protected:
+  size_t n_trees_;
+  int64_t n_targets_or_classes_;
+  POST_EVAL_TRANSFORM post_transform_;
+  const std::vector<OTYPE>* base_values_;
+  OTYPE origin_;
+  bool use_base_values_;
+
+ public:
+  inline _Aggregator(size_t n_trees,
```

Comment:
>inline [](start = 2, length = 6)

nit: 'inline' isn't needed here

[https://en.cppreference.com/w/cpp/language/inline](https://en.cppreference.com/w/cpp/language/inline)
"A function defined entirely inside a class/struct/union definition, whether it's a member function or a non-member friend function, is implicitly an inline function."

Reply:
fixed